crypto/thr_id.c

```diff
@@ -199,6 +204,8 @@ void CRYPTO_THREADID_current(CRYPTO_THREADID *id)
     /* Else pick a backup */
 #if defined(OPENSSL_SYS_WIN32)
     CRYPTO_THREADID_set_numeric(id, (unsigned long)GetCurrentThreadId());
+#elif defined(__unix__) || defined(unix)
+    CRYPTO_THREADID_set_numeric(id, (unsigned long)pthread_self());
```

Comment:
My OS X errno.h #defines it to (*__error()) akin to my linux (*errno_location()) (under appropriate circumstances).  Is your OS X test where it doesn't work compiling as threaded code?